src/API/Management/ClientGrants.php

```diff
@@ -2,75 +2,193 @@
+     *      - "client_id" to return an object.
+     *      - "include_totals" to return an object.
+     * @param null|integer $page     The page number, zero based.
+     * @param null|integer $per_page The amount of entries per page.
      *
-     * @param  string      $id
-     * @param  null|string $audience
      * @return mixed
+     *
+     * @throws \Exception Thrown by the Guzzle HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Client_Grants/get_client_grants
      */
-    public function get($id, $audience = null)
+    public function getAll(array $params = [], $page = null, $per_page = null)
     {
-        $request = $this->apiClient->get()
-        ->addPath('client-grants');
+        if (null !== $page) {
+            $params['page'] = abs(intval($page));
+        }
 
-        if ($audience !== null) {
-            $request = $request->withParam('audience', $audience);
+        if (null !== $per_page) {
+            $params['per_page'] = abs(intval($per_page));
         }
 
-        return $request->call();
+        return $this->apiClient->method('get')
+            ->addPath('client-grants')
+            ->withDictParams($params)
+            ->call();
+    }
+
+    /**
+     * Get Client Grants by audience.
+     * Required scope: "read:client_grants"
+     *
+     * @param string       $audience API Audience to filter by.
+     * @param null|integer $page     The page number, zero based.
+     * @param null|integer $per_page The amount of entries per page.
+     *
+     * @return mixed
+     *
+     * @throws CoreException Thrown when $audience is empty or not a string.
+     * @throws \Exception Thrown by the Guzzle HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Client_Grants/get_client_grants
+     */
+    public function getByAudience($audience, $page = null, $per_page = null)
+    {
+        if (empty($audience) || ! is_string($audience)) {
+            throw new CoreException('Empty or invalid "audience" parameter.');
+        }
+
+        return $this->getAll(['audience' => $audience], $page, $per_page);
+    }
+
+    /**
+     * Get Client Grants by Client ID.
+     * Required scope: "read:client_grants"
+     *
+     * @param string       $client_id Client ID to filter by.
+     * @param null|integer $page      The page number, zero based.
+     * @param null|integer $per_page  The amount of entries per page.
+     *
+     * @return mixed
+     *
+     * @throws CoreException Thrown when $client_id is empty or not a string.
+     * @throws \Exception Thrown by the Guzzle HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Client_Grants/get_client_grants
+     */
+    public function getByClientId($client_id, $page = null, $per_page = null)
```

Comment:
Is this convenience method also in other SDKs?
/cc @lbalmaceda

Reply:
Not that I know of but I don't think that's a good reason not to add it here. Make the developer's life easier!

Reply:
Just curious, it's always good to create more convenience for the end user, balanced with maintainability.